src/createDynamicHostResolver.js

```diff
@@ -0,0 +1,111 @@
+/***************************************************************************************
+ * (c) 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ ****************************************************************************************/
+
+var window = require('@adobe/reactor-window');
+
+module.exports = function (turbineEmbedCode, cdnAllowList, debugController) {
+  // even an empty list is flagging to us that we're trying to enforce dynamic
+  var isDynamicEnforced = Array.isArray(cdnAllowList);
+  var embedPattern = /^https?:\/\/.*/;
+
+  // TODO: web only? I think embedded TVs wouldn't have
+  //  __satellite.container.dynamicEnforced turned on
+  var turbineUrl = document.createElement('a');
+  turbineUrl.href = turbineEmbedCode;
+  if (
+    (!embedPattern.test(turbineEmbedCode) || !turbineUrl.host) &&
+    isDynamicEnforced === true
+  ) {
+    var missingEmbedCodeError = new Error(
+      'Unable to find the Library Embed Code for Dynamic Host Resolution.'
+    );
+    missingEmbedCodeError.code = 'dynamic_host_resolver_constructor_error';
+    throw missingEmbedCodeError;
+  }
+
+  if (isDynamicEnforced && cdnAllowList.indexOf(turbineUrl.hostname) === -1) {
+    var dynamicDeniedError = new Error(
+      'This library is not authorized for this domain. ' +
+        'Please contact your CSM for more information.'
+    );
+    dynamicDeniedError.code = 'dynamic_host_not_allowed';
+    throw dynamicDeniedError;
+  }
+
+  var shouldAugment = Boolean(isDynamicEnforced && turbineUrl);
```

Comment:
Is there any situation when `turbineUrl` would not be `truthy`?

Reply:
I can clean this up. The idea to stop processing came later in development so now that check doesn't necessarily make sense anymore. This was also before using `document.createElement('a');` to support IE.